problems/arraysWithLoops.js

```diff
@@ -25,39 +39,62 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+function shoutForOfLoop(array) {
+  let array2 = []
+  for (let element of array) array2.push(element + "!")
+  return array2
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+function sumArray(nums) {
+  let sum = 0
+  for (let i = 0; i <= nums.length - 1; i++) sum += nums[i];
+  return sum
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+function oddArray(nums) {
+  let sum = []
+  for (let i = 0; i <= nums.length - 1; i++)
+  if (nums[i] % 2 === 1) sum.push(nums[i]);
+  return sum
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+function evenArray(nums) {
+  let sum = []
+  for (let i = 0; i <= nums.length - 1; i++)
+  if (nums[i] % 2 === 0) sum.push(nums[i]);
+  return sum
+}
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+function findSmallest(nums) {
+  let numSmall = []
+  for (let i = 0; i < nums.length; i++) {
+    if (numSmall[i] < nums)
+    return numSmall
+}}
```

Comment:
Lets make sure we are using the curly brackets where appropriate. Rewrite your if statement to do something with the current smallest number